### PR TITLE
Heading snapping bug

### LINKED ISSUE #3: Auto-locking heading
Implemented in frc937/swerving-learning-box on branch heading-locking. Not working there, not sure why. Maybe reference for implementation?

### PR DESCRIPTION
**Pull request by Jack-Haefele**
closes #3

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 1]
```diff
@@ -77,6 +77,15 @@ public void enterXMode() {
     drive.lockPose();
   }
 
+  /**
+   * Set the heading correction capabilitys of YAGSL.
```
Comment (willitcode) [inline src/main/java/frc/robot/subsystems/Drive.java:81]: "capabilities"

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 2]
```diff
@@ -77,6 +77,15 @@ public void enterXMode() {
...
+   *
+   * @param state SwerveDrive.headingCorrection state
```
Comment (willitcode) [inline src/main/java/frc/robot/subsystems/Drive.java:83]: "{@link SwerveDrive.headingCorrection}"

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 3]
```diff
@@ -77,6 +77,15 @@ public void enterXMode() {
     drive.lockPose();
   }
 
+  /**
+   * Set the heading correction capabilitys of YAGSL.
+   *
```
Comment (willitcode) [inline src/main/java/frc/robot/subsystems/Drive.java:82]: I'd add another paragraph here that says something like "Should only be enabled when heading correction capabilities are in use"

**Review by willitcode — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 1 above)
Comment (Jack-Haefele) [reply src/main/java/frc/robot/subsystems/Drive.java:81]: >:(

**Review by willitcode — APPROVED** [review]

**Pull request merged**